user_guide_src/source/libraries/validation.rst

```diff
@@ -992,3 +995,7 @@ is_image                Yes         Fails if the file cannot be determined to be
 ======================= ========== ============================================= ===================================================
 
 The file validation rules apply for both single and multiple file uploads.
+
+.. note:: Only rules specifically created for file validation (like the ones listed in the table above) can be used to validate files.
+    Therefore, adding any general rules, like ``permit_empy``, to file validation rules array or string will prevend all rules for the
+    from file from being processed during validation.
```

Comment:
I don't get well "prevend all rules for the from file from being processed during validation."

Reply:
How about this?

```suggestion
    Therefore, if you add any general rules, like ``permit_empy`` to file validation rules array or string,
    the file validation will not work correctly.
```

Reply:
Well, if you set rules 'permit_empty|max_size[profile_img,1024]|mime_in[profile_img,image/jpeg]', you can uplood 3mb pdf's, and validation will pass.

Reply:
Yep, but I cannot believe I messed up my English that much... sorry.

Reply:
``permit_empty`` (or other general rules) checks `$_POST` array, and the field for the file upload does not exist, so the validation passes.
However, most rules do not permit empty, and `$_POST` has no field for it, so if set, the validation will fail.

I don't know how to explain it briefly.

Reply:
Ah, it's complicated. I was also thinking about filing a bug report (it is not good that adding a stray rule disables the other rules), but now I suspect that it is more of a limitation of validation than a bug...